src/poppler/document.py

```diff
@@ -277,18 +280,22 @@ def unlock(self, owner_password, user_password):
         return self._document.unlock(owner_password, user_password)
 
 
-def load_from_file(file_name, owner_password=None, user_password=None):
+def load_from_file(
+    file_name: str, owner_password: str = "", user_password: str = ""
+) -> Document:
+    if not isinstance(file_name, str):
+        raise TypeError("file_name MUST be str")
```

Comment:
I do not agree with this. You may pass a Path object to the function. This is why I wrote the function to accept any object that is convertible to string.

Reply:
then perhaps `Union[str, Path]` and checking `isinstance(filename, (Path, str))`?

Reply:
Yes for annotating as `Union[str, Path]`. No for the `isinstance` check. The philosophy in Python is to allow duck typing. If it looks like a duck and it quacks like a duck, then it should be a duck.

Reply:
Okay will update the PR soon